tests/conftest.py

```diff
@@ -258,18 +258,23 @@ def microvm_factory(request, record_property, results_dir):
     uvm_factory = MicroVMFactory(fc_binary_path, jailer_binary_path)
     yield uvm_factory
 
-    # if the test failed, save fc.log in test_results for troubleshooting
+    # if the test failed, save important files from the root of the uVM into `test_results` for troubleshooting
     report = request.node.stash[PHASE_REPORT_KEY]
     if "call" in report and report["call"].failed:
         for uvm in uvm_factory.vms:
-            if not uvm.log_file.exists():
-                continue
-            dst = results_dir / uvm.id / uvm.log_file.name
-            dst.parent.mkdir()
-            shutil.copy(uvm.log_file, dst)
-            if uvm.metrics_file.exists():
-                dst = results_dir / uvm.id / uvm.metrics_file.name
-                shutil.copy(uvm.metrics_file, dst)
+            uvm_data = results_dir / uvm.id
+            uvm_data.mkdir()
+
+            uvm_root = uvm.path / "root"
+            items_to_copy = [
+                item
+                for item in os.listdir(uvm_root)
+                if os.path.isfile(uvm_root / item) and not item.endswith("sock")
+            ]
+            for item in items_to_copy:
+                src = uvm_root / item
+                dst = uvm_data / item
+                shutil.copy(src, dst)
```

Comment:
Nit: a bit shorter/easier to read:

```
for uvm_file in uvm.chroot().rglob("*"):
    if not uvm_file.is_file():
        continue
    dst = uvm_data / uvm_file.name
    shutil.copy(uvm_file, dst)
```

Also I don't is needed to check for ".sock" or ".socket" since those are not regular files.

Reply:
Updated